### PR TITLE
feat: "preview" handlebar for differentiating preview & print

### PR DESCRIPTION
**Pull request by Zangetsu101**

### CODE REVIEW

File: packages/client/src/views/PrintCertificate/usePrintableCertificate.ts [hunk 1]
```diff
@@ -156,29 +156,26 @@ export const usePrintableCertificate = (declarationId: string) => {
...
+        draft.event,
+        eventDate,
+        registeredDate,
+        offlineData
+      )
+      certificate.payments = {
+        type: 'MANUAL' as const,
+        amount: Number(paymentAmount),
+        outcome: 'COMPLETED' as const,
+        date: new Date().toISOString()
```
Comment (Zangetsu101) [inline packages/client/src/views/PrintCertificate/usePrintableCertificate.ts:169]: `isFreeOfCost` used to check if the amount was equal to 0

File: packages/client/src/forms/register/mappings/query/registration-mappings.ts [hunk 2]
```diff
@@ -138,7 +139,7 @@ export const localPhoneTransformer =
   }
 
 const getUserFullName = (history: History): string => {
-  return history?.user ? getUserName(history.user) : ''
+  return history?.user?.name ? createNamesMap(history.user.name)[LANG_EN] : ''
```
Comment (rikukissa) [inline packages/client/src/forms/register/mappings/query/registration-mappings.ts:142]: Is this a bug fix to some issue in the flow?

File: packages/client/src/views/PrintCertificate/usePrintableCertificate.ts [hunk 3]
```diff
@@ -156,29 +156,26 @@ export const usePrintableCertificate = (declarationId: string) => {
...
       }
     }
 
+    const svg = await compileSvg(
+      offlineData.templates.certificates[draft.event].definition,
+      { ...draft.data.template, print: true },
```
Comment (rikukissa) [inline packages/client/src/views/PrintCertificate/usePrintableCertificate.ts:175]: Reckon we should do an explicit `print: false` or even `print: false, preview: true` for preview? That got me also thinking if we should only have `preview: boolean` which could be more descriptive than `print`. What do you think?

File: packages/client/src/views/PrintCertificate/usePrintableCertificate.ts [hunk 4]
```diff
@@ -123,29 +116,36 @@ export const usePrintableCertificate = (declarationId: string) => {
...
   useEffect(() => {
-    if (declaration)
-      getPDFTemplateWithSVG(offlineData, declaration, state).then((svg) => {
-        const svgWithFonts = addFontsToSvg(
-          svg.svgCode,
-          offlineData.templates.fonts ?? {}
-        )
-        setSvg(svgWithFonts)
-      })
+    const certificateTemplate =
```
Comment (rikukissa) [inline packages/client/src/views/PrintCertificate/usePrintableCertificate.ts:126]: Not anything you need to action on but to me this whole hook feels like it's doing way too many thing starting from the fact that it fetches declarations by id and doing printing and certifying as well. Would be great to split this up in the near future

**Review by rikukissa — APPROVED** [review]

File: packages/client/src/forms/register/mappings/query/registration-mappings.ts (hunk 2 above)
Comment (Zangetsu101) [reply packages/client/src/forms/register/mappings/query/registration-mappings.ts:142]: It's not actually. `getUserName` was one of the transformer functions related to pdfmake that I removed in this PR. So had to refactor

File: packages/client/src/views/SysAdmin/Config/Certificates/Certificates.tsx [hunk 5]
```diff
@@ -202,26 +181,24 @@ class CertificatesConfigComponent extends React.Component<Props, State> {
...
+    svgTemplate: string,
     svgFilename: string
   ) {
+    const dummyTemplateData = getDummyCertificateTemplateData(
+      event,
+      this.props.registerForm
+    )
+    const compiledSvgPromise = compileSvg(
+      svgTemplate,
+      dummyTemplateData,
```
Comment (rikukissa) [inline packages/client/src/views/SysAdmin/Config/Certificates/Certificates.tsx:193]:
```suggestion
      { ...dummyTemplateData, preview: false },
```

File: packages/client/src/views/PrintCertificate/usePrintableCertificate.ts [hunk 6]
```diff
@@ -156,29 +158,26 @@ export const usePrintableCertificate = (declarationId: string) => {
...
+        type: 'MANUAL' as const,
+        amount: Number(paymentAmount),
+        outcome: 'COMPLETED' as const,
+        date: new Date().toISOString()
       }
     }
 
+    const svg = await compileSvg(
+      offlineData.templates.certificates[draft.event].definition,
+      draft.data.template,
```
Comment (rikukissa) [inline packages/client/src/views/PrintCertificate/usePrintableCertificate.ts:177]:
```suggestion
      { ...draft.data.template, preview: false },
```

**Comment by rikukissa** [comment]
I've tested this locally with a custom certificate where "PREVIEW"  text is controlled with a `ifCond` clause
```svg
{{#ifCond preview '===' true}}
  <text fill="red" xml:space="preserve" style="white-space: pre" font-family="Noto Sans" font-size="15.52" font-weight="bold" letter-spacing="3.88px"><tspan x="200.256" y="20.002">PREVIEW</tspan></text>
{{/ifCond}}
```

**Preview:**
image

**Print:**
image

Looks like it's working as it should 👍

**Review by rikukissa — APPROVED** [review]

**Pull request merged**